android/src/main/java/com/pkmnapps/nearby_connections/NearbyConnectionsPlugin.java

```diff
@@ -96,29 +105,29 @@ public void onMethodCall(MethodCall call, final Result result) {
                 result.success(gps_enabled || network_enabled);
                 break;
             case "enableLocationServices":
-                activity.startActivity(new Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS));
+                locationEnabler.requestLocationEnable(result);
```

Comment:
I think there could be cases where `result.success`/`result.error` is never called and this will go unnoticed. You can probably add callbacks (`success`/`error`) to this function, and set `result.success`/`result.error` here using these callbacks.

Reply:
callback added to task, please have a look